### PR TITLE
Update website to display new `workload` and `suite` changes correctly

### PR DESCRIPTION
**Pull request by Harshil2107**
This change has 3 parts:
- Updating the gem5 resources schema to now include suites and new changes to the workloads JSON object.
- Fixing a bug that was introduced due to the change made to workloads where new workloads with the new JSON format were not displayed on the website.
- Updating the tests that would fail due to addition on the new category "suite".

### CODE REVIEW

**Comment by powerjg** [comment]
Can you give some context to this change with the commit message or the PR message? I don't know what I'm reviewing.

File: package.json [hunk 1]
```diff
@@ -50,7 +50,7 @@
     "babel-jest": "^29.5.0",
     "cypress": "^12.10.0",
     "identity-obj-proxy": "^3.0.0",
-    "jest": "^29.5.0",
+    "jest": "^29.7.0",
```
Comment (powerjg) [inline package.json:53]: This change seems unrelated

File: public/all-gem5-resources-schemas/gem5-resources-schema-23.0.json
Comment (powerjg) [inline public/all-gem5-resources-schemas/gem5-resources-schema-23.0.json:1]: This change seems unrelated to "workload bug"

File: public/gem5-resources-schema.json
Comment (powerjg) [inline public/gem5-resources-schema.json:1]: Why are there 2-3 copies of this file?

File: package-lock.json
Comment (powerjg) [inline package-lock.json:1]: Should this file be committed to git?

**Review by powerjg — COMMENTED** [review]
Seems like there's a lot more than just "fix workload bug" in this PR. Please update the PR description and/or commits to make it more clear how to review this.

File: package.json (hunk 1 above)
Comment (Harshil2107) [reply package.json:53]: While running the tests I noticed that new version for jest was out, so I added it. Should I make a separate PR or is it ok to not update the version?

File: public/all-gem5-resources-schemas/gem5-resources-schema-23.0.json
Comment (Harshil2107) [reply public/all-gem5-resources-schemas/gem5-resources-schema-23.0.json:1]: This is related in the sense that this change contains new schema, we were waiting on updating website till the release. So I added the bug fix as the bug was introduced due to the updated schema for workloads. So I added the bug fix on top of the schema change.

File: public/gem5-resources-schema.json
Comment (Harshil2107) [reply public/gem5-resources-schema.json:1]: There are 2 copies of the latest schema on in the test folder and one in public. The reason for having two separate copies is to catch the change in schema and so we have to update the tests accordingly.

There is also a new folder which will keep older schemas in case we need to refer to them.

File: package-lock.json
Comment (Harshil2107) [reply package-lock.json:1]: Yes as this file makes sure that if anyone else forks or commits to this repo, all the dependencies are the same.

**Comment by Harshil2107** [comment]
>  Can you give some context to this change with the commit message or the PR message? I don't know what I'm reviewing.

I updated the PR message.

File: package-lock.json
Comment (BobbyRBruce) [reply package-lock.json:1]: This is annoying NPB thing. It's a way to guarantee all the dependencies(in particular the transative dependencies) are kept constant between installations (it's slightly different to `package.json`). I find it very weird and annoying but it's explained here: https://stackoverflow.com/questions/44297803/what-is-the-role-of-the-package-lock-json

File: public/gem5-resources-schema.json
Comment (powerjg) [reply public/gem5-resources-schema.json:1]: Can you have a symlink from the tests to the public one?

File: package.json (hunk 1 above)
Comment (powerjg) [reply package.json:53]: That's fine, but it should be it's own commit, at least.

File: public/all-gem5-resources-schemas/gem5-resources-schema-23.0.json
Comment (powerjg) [reply public/all-gem5-resources-schemas/gem5-resources-schema-23.0.json:1]: I think a better title to the PR would be "adapt website to new schema" and one of the commits could be "fix bug <give more information."

Remember, the people reviewing the code can't read your mind, and they likely will forget all of the information that you've told them previously. Each PR should be completely stand alone and someone should be able to understand why you made each change in the PR with the information in the PR.

**Review by BobbyRBruce — COMMENTED** [review]
"Fix workload bug" doesn't really mean much to me here. Can we have something a bit more descriptive?

I'm honestly a bit lost as to what our schema means anymore. We're changing it here and I don't know why and what it means for gem5-resources. Can we just update this whenever we want or are there procedures we need to follow?

**Comment by Harshil2107** [comment]
I think we help off on the schema update till release. I dont know if we have a strong reason part for wanting at most one schema change per release?

In this update the change to schema are the following:

- Added suites as a category.
- Updated the workload schema so that now the `resources` field is 
```
resources: {
  binary: { 
                  'id': id,
                  'resource_version':"1.0.0"
                  },
  .
  .
  .
  } 
```
  The older `resources` filed was:
```
resources: {
  binary: id,
  .
  .
  .
  }
```

File: public/gem5-resources-schema.json
Comment (Harshil2107) [reply public/gem5-resources-schema.json:1]: Yes, I think we can. Our thought process when adding separate files was that if there is a major schema change like in our case adding a new category then the cypress test will fail. This was to make sure that such schema changes are caught and then tests have to be modified, this also serves as a way to double check that schema does not break anything else.

**Review by BobbyRBruce — APPROVED** [review]
I still have a hard time wrapping my head around what the purpose is anymore for the schema. I feel we have deviated from our original goal and i worry we're just updating it as we need it in a hacky way.

However, if this fixes the bug then we should get it in and we can discuss procedure later.
